compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3003,76 +3011,315 @@ private BLangExpression createConditionFromMatchPatterns(List<BLangMatchPattern>
+                    symTable.anydataOrReadonly,
+                    fieldBasedAccessExpr, matchPatternPos);
+            mainBlockStmt.addStatement(tempVarDef);
+            BLangSimpleVarRef tempVarRef = ASTBuilderUtil.createVariableRef(matchPatternPos, tempVarDef.var.symbol);
+
+            BLangExpression varCheckCondition = createConditionForNamedArgMatchPattern(matchPattern, tempVarRef);
+            if (i == 0) {
+                condition = varCheckCondition;
+                continue;
+            }
+            condition = ASTBuilderUtil.createBinaryExpr(matchPatternPos, condition, varCheckCondition,
+                    symTable.booleanType, OperatorKind.AND, (BOperatorSymbol) symResolver
+                            .resolveBinaryOperator(OperatorKind.AND, symTable.booleanType, symTable.booleanType));
+
+        }
+
+        BLangBlockStmt tempBLock = ASTBuilderUtil.createBlockStmt(pos);
+        tempBLock.addStatement(successResult);
+        BLangIf ifStmtForFieldMatchPatterns = ASTBuilderUtil.createIfElseStmt(pos, condition, tempBLock, null);
+        mainBlockStmt.addStatement(ifStmtForFieldMatchPatterns);
+
+        BLangStatementExpression statementExpression = ASTBuilderUtil.createStatementExpression(mainBlockStmt,
+                resultVarRef);
+        statementExpression.type = symTable.booleanType;
+
+        return statementExpression;
+    }
+
+    private BLangExpression createConditionForNamedArgMatchPattern(BLangMatchPattern matchPattern,
+                                                                   BLangSimpleVarRef matchExprVarRef) {
+
+        return createVarCheckCondition(matchPattern, matchExprVarRef);
+    }
+
+    private BLangExpression createConditionForErrorCauseMatchPattern(BLangErrorCauseMatchPattern errorCausePattern,
+                                                                     BLangSimpleVarRef matchExprVarRef) {
+
+        if (errorCausePattern.simpleMatchPattern != null) {
+            return createConditionForSimpleMatchPattern(errorCausePattern.simpleMatchPattern, matchExprVarRef);
+        }
+        return createConditionForErrorMatchPattern(errorCausePattern.errorMatchPattern, matchExprVarRef);
+    }
+
+    private BLangExpression createConditionForErrorMessageMatchPattern(BLangErrorMessageMatchPattern errorMsgPattern,
+                                                                       BLangSimpleVarRef matchExprVarRef) {
+
+        return createConditionForSimpleMatchPattern(errorMsgPattern.simpleMatchPattern, matchExprVarRef);
+    }
+
+    private BLangExpression createConditionForSimpleMatchPattern(BLangSimpleMatchPattern simpleMatchPattern,
+                                                                 BLangSimpleVarRef matchExprVarRef) {
+
+        if (simpleMatchPattern.wildCardMatchPattern != null) {
+            return createVarCheckCondition(simpleMatchPattern.wildCardMatchPattern, matchExprVarRef);
+        } else if (simpleMatchPattern.constPattern != null) {
```

Comment:
No need to use the else-if clause, we can use the if clause

Reply:
Done